clang/lib/ExtractAPI/Serialization/SerializerBase.cpp

```diff
@@ -16,4 +16,5 @@
 
 using namespace clang::extractapi;
 
-void APISerializer::serialize(llvm::raw_ostream &os) {}
+template <typename Derived>
+void APISerializer<Derived>::serialize(llvm::raw_ostream &os) {}
```

Comment:
Unfortunately making the class a function template means you have to move the implementation into the header file.